ui/packages/catalog-ui-search/src/main/webapp/react-component/container/map-actions/map-actions.tsx

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+
+import { hot } from 'react-hot-loader'
+import * as React from 'react'
+const wreqr = require('../../../js/wreqr.js')
+const _ = require('underscore')
+
+type Props = {
+  model: Backbone.Model
+}
+
+type State = {
+  currentOverlayUrl: string
+}
+
+export default hot(module)(
+  class MapActions extends React.Component<Props, State> {
+    constructor(props: Props) {
+      super(props)
+
+      this.state = {
+        currentOverlayUrl: this.props.model.get('currentOverlayUrl'),
+      }
+    }
+
+    getActions = () => {
+      return this.props.model.get('actions')
+    }
+
+    getMapActions = () => {
+      return this.getActions().filter(function(action: any) {
+        return action.get('id').indexOf('catalog.data.metacard.map.') === 0
+      })
+    }
+
+    getOverlayActions = () => {
+      let modelOverlayActions = this.getActions().filter(function(action: any) {
```

Comment:
These two functions are almost doing the same thing.
```javascript
return this.getActions().filter(function(action: any) {
  return action.get('id').indexOf('catalog.data.metacard.map.') === 0
})
```

```javascript
let modelOverlayActions = this.getActions().filter(function(action: any) {
return (
  action.get('id').indexOf('catalog.data.metacard.map.overlay.') === 0
)
```

You can replace with something like this
```javascript
getActionsWithIdPrefix(actions, id){
  return actions.filter((action: any) => action.get('id').indexOf(id) === 0
}
```

and then use it like this

```javascript
let modelOverlayActions = getActionsWithIdPrefix(this.getActions(), 'catalog.data.metacard.map.overlay.')
```